packages/server/src/schema/context.ts

```diff
@@ -1,16 +1,15 @@
 import { Slice } from '@effection/atom';
 import { Task } from 'effection';
-import { v4 } from 'uuid';
 import { SimulationState } from '../interfaces';
 
 export class SimulationContext {
-  constructor(private scope: Task, private simulations: Slice<Record<string, SimulationState>>) {}
+  constructor(private scope: Task, private simulations: Slice<Record<string, SimulationState>>, private newid: () => string) {}
 
-  async createSimulation(using: string | string[], uuid?: string): Promise<SimulationState> {
+  async createSimulation(using: string | string[]): Promise<SimulationState> {
     let simulators = ([] as string[]).concat(using);
-    let { scope, simulations } = this;
+    let { scope, simulations, newid } = this;
 
-    let id = uuid || v4();
+    let id = newid();
```

Comment:
does this guarantees the same uuids each time if I do:

```ts
for (const i = o; i < 5 ; i++) {
  console.log(newid());
``

Reply:
Yep. here's a quick test:

```ts
import { createFaker } from './faker';

[1,2,3].forEach(trial => {
  console.log(`trial ${trial}`);
  let faker = createFaker(100);
  [1,2,3,4,5].forEach((i) => {
    console.log(`  ${i}. ${faker.datatype.uuid()}`);
  });
});
```

```
trial 1
  1. 8a4668d2-0212-4a3d-b2d9-1e938281034f
  2. fc221d44-a6bf-4ade-9429-52084ac20997
  3. 96176607-e0fc-408e-a98b-fa2910d318dc
  4. d4140d0f-5ea5-4091-9e5c-2c3a09826991
  5. a523e8f6-9f62-45b3-84b3-d28f0fe95b85
trial 2
  1. 8a4668d2-0212-4a3d-b2d9-1e938281034f
  2. fc221d44-a6bf-4ade-9429-52084ac20997
  3. 96176607-e0fc-408e-a98b-fa2910d318dc
  4. d4140d0f-5ea5-4091-9e5c-2c3a09826991
  5. a523e8f6-9f62-45b3-84b3-d28f0fe95b85
trial 3
  1. 8a4668d2-0212-4a3d-b2d9-1e938281034f
  2. fc221d44-a6bf-4ade-9429-52084ac20997
  3. 96176607-e0fc-408e-a98b-fa2910d318dc
  4. d4140d0f-5ea5-4091-9e5c-2c3a09826991
  5. a523e8f6-9f62-45b3-84b3-d28f0fe95b85
```

Reply:
nice!